### DISCUSSION TITLE
BIRT 480 and WebSphere Chart reports fail on javax/xml/datatype/XMLGregorianCalendar.getXMLSchemaType()

### DISCUSSION THREAD

**Discussion by dthunde** [eclipse-birt/birt#1066] [Q&A]
Have WebSphere and BIRT servlet working for standard reports.  Reports with charting fail on javax library in WebSphere and BIRT with: at com.ibm.ws.util.ThreadPool$Worker.run(ThreadPool.java:1892)
+6477  Caused by: java.lang.LinkageError: loading constraint violation when overriding method "javax/xml/datatype/XMLGregorianCalendar.getXMLSchemaType()Ljavax/xml/namespace/QName
;" during creation of class "org/eclipse/emf/ecore/xml/type/internal/XMLCalendar": loader"com/ibm/ws/classloader/CompoundClassLoader@4eaded11" of class
"org/eclipse/emf/ecore/xml/type/internal/XMLCalendar" and loader "com/ibm/oti/vm/BootstrapClassLoader@42db59b4" of class "javax/xml/datatype/XMLGregorianCalendar" have different types for the method signature

Tried removing the XMLCalendar class from the Eclipse library.  The method getXMLSchemaType has no parameters.  It appears the Eclipse libarary and the javax library in WebSphere return different data types.  I see the Eclipse library is at Java version 5.  I can't seem to find where the javax library is in WebSphere.  Can find no reference to getXMLSchemaType having a function change.

Chart reports run in batch mode, referring only to the library supplied by BIRT.
![javasxXML](https://user-images.githubusercontent.com/103052009/161778707-d9a88c22-1e99-4542-ac2a-b2bbd2d6a3bc.PNG)
![javaXeclipse](https://user-images.githubusercontent.com/103052009/161778733-fd338271-da24-4c7e-b023-544a15b5377a.PNG)

**Comment by claesrosell** [comment]
From the error message I get a feeling that this is caused by two different versions of javax.xml.namespace.QName. It may be the same version loaded by two different ClassLoaders as well. I am not that familiar with WebSphere.

Can you test with Birt 4.9.0?

**Comment by dthunde** [comment]
Thanks, I didn't think of that.  I will try removing the namespace class in BIRT library, referring to the library in WebSphere.  I suspect I have two conflicting versions.  WebSphere version doesn't return a Qname class and suspect code will fail.

Do you know if Birt 4.9 can by used with JDK 8.0?

I will try Birt 4.9, but would hate to find it uses a higher JDK version.  I think the install checks this.

We are stuck on JDK 8.0 for a long time.

**Comment by hvbtup** [comment]
BIRT 4.9 requires Java 11 or newer.
